PSGSuite/Public/Calendar/Get-GSCalendarSubscription.ps1

```diff
@@ -68,7 +80,20 @@ function Get-GSCalendarSubscription {
             try {
                 Write-Verbose "Getting subscribed calendar list for user '$User'"
                 $request = $service.CalendarList.List()
-                $request.Execute() | Select-Object -ExpandProperty Items
+                If ($ShowDeleted -eq $true){
+                    $request.$ShowDeleted = $true
+                }
+                If ($ShowHidden -eq $true){
+                    $request.$ShowHidden = $true
+                }
+                $raw = $request.Execute()
+                $ItemList = [object[]]($raw | Select-Object -ExpandProperty Items)
```

Comment:
Any new `List()` requests should follow the same patterns in place for others, ideally. This is true for both iterating through the paged results as well as adding additional properties to the `$request` object. Here's a good example of a `List()` request in place that already has this structure: https://github.com/scrthq/PSGSuite/blob/ac1a0c5273eb7a284a14ae5d82b1a1360af1a4ea/PSGSuite/Public/Tasks/Get-GSTask.ps1#L116..L143

```powershell
try {
    Write-Verbose "Getting all Tasks from Tasklist '$Tasklist' for user '$User'"
    $request = $service.Tasks.List($Tasklist)
    foreach ($key in $PSBoundParameters.Keys | Where-Object {$request.PSObject.Properties.Name -contains $_}) {
        switch ($key) {
            Tasklist {}
            {$_ -in @('CompletedMax','CompletedMin','DueMax','DueMin','UpdatedMin')} {
                $request.$key = ($PSBoundParameters[$key]).ToString('o')
            }
            default {
                if ($request.PSObject.Properties.Name -contains $key) {
                    $request.$key = $PSBoundParameters[$key]
                }
            }
        }
    }
    if ($PSBoundParameters.Keys -contains 'PageSize') {
        $request.MaxResults = $PSBoundParameters['PageSize']
    }
    [int]$i = 1
    do {
        $result = $request.Execute()
        $result.Items | Add-Member -MemberType NoteProperty -Name 'User' -Value $User -PassThru
        $request.PageToken = $result.NextPageToken
        [int]$retrieved = ($i + $result.Items.Count) - 1
        Write-Verbose "Retrieved $retrieved Tasks..."
        [int]$i = $i + $result.Items.Count
    }
    until (!$result.NextPageToken)
}
catch {
    if ($ErrorActionPreference -eq 'Stop') {
        $PSCmdlet.ThrowTerminatingError($_)
    }
    else {
        Write-Error $_
    }
}
```